src/xenia/kernel/XLiveAPI.cpp

```diff
@@ -1057,23 +1057,28 @@ XONLINE_SERVICE_INFO XLiveAPI::GetServiceInfoById(uint32_t serviceId) {
 }
 
 void XLiveAPI::SessionJoinRemote(uint64_t sessionId,
-                                 const std::vector<std::string> xuids) {
+                                 std::map<std::string, bool> members) {
   std::string endpoint = fmt::format("title/{:08X}/sessions/{:016x}/join",
                                      kernel_state()->title_id(), sessionId);
 
   Document doc;
   doc.SetObject();
 
-  Value xuidsJsonArray(kArrayType);
+  Value xuidsJsonArray = Value(kArrayType);
+  Value privateSlotsJsonArray = Value(kArrayType);
 
-  for each (const auto xuid in xuids) {
-    Value value;
-    value.SetString(xuid.c_str(), 16, doc.GetAllocator());
+  for (const auto& [xuid, is_private] : members) {
```

Comment:
Why xuid is stored as a string? It might be better to store it as a uint64_t and ignore converting there? (and calculating hash for each string?)